libs/browser/shared/hooks/src/lib/__test__/useKeyboardMoveFocus.spec.ts

```diff
@@ -0,0 +1,204 @@
+      });
+
+      expect(seq).toEqual(result);
+    },
+  );
+
+  it.each`
+    currIndex | columns | infinite | result
+    ${1}      | ${5}    | ${true}  | ${[11, 2, 6, 0, 0, 18, 1, 2, 3, 12, 1, 0]}
+    ${18}     | ${6}    | ${false} | ${[12, null, null, null, 0, null, 1, 2, 3, 12, 1, 0]}
+  `(
+    '[useKeyboardMoveFocus] should call onFocusedIndexChange correctly with disabled options',
+    ({ currIndex, columns, infinite, result }) => {
+      const keys = [
+        ['ArrowUp'],
+        ['ArrowRight'],
+        ['ArrowDown'],
+        ['ArrowLeft'],
+        ['Home'],
+        ['End'],
+        ['1'],
+        ['2'],
+        ['3'],
+        ['1', '2'],
+        ['1', '9'],
+        ['0', '2'],
+      ];
+
+      const disabledDict = <Record<number, boolean>>{ 16: true, 19: true };
+
+      const seq = getFocusSequence(keys, {
+        focusedIndexRef: renderHook(() => currIndex).result,
+        options,
+        columns,
+        infinite,
+        onFocusedIndexChange: mockOnFocusedIndexChangeFn,
+        getOptionDisabled: (option) => !!disabledDict[option],
+      });
+
+      expect(seq).toEqual(result);
+    },
+  );
+
+  it.each`
+    currIndex | columns | infinite | result
+    ${2}      | ${5}    | ${true}  | ${18}
+    ${15}     | ${5}    | ${true}  | ${15}
+    ${16}     | ${6}    | ${false} | ${15}
+  `(
+    '[useKeyboardMoveFocus] getPrevFocusableOption should return previous focusable option',
+    ({ currIndex, columns, infinite, result }) => {
+      const disabledDict = <Record<number, boolean>>{
+        0: true,
+        1: true,
+        2: true,
+        16: true,
+        19: true,
+      };
+
+      const { getPrevFocusableOption } = _useKeyboardMoveFocus({
+        focusedIndexRef: renderHook(() => currIndex).result,
+        options,
+        columns,
+        infinite,
+        onFocusedIndexChange: (_) => {},
+        getOptionDisabled: (option) => !!disabledDict[option],
+      });
+      expect(getPrevFocusableOption()).toEqual(result);
+    },
+  );
+
+  it.each`
+    currIndex | columns | infinite | result
+    ${0}      | ${5}    | ${true}  | ${3}
+    ${16}     | ${5}    | ${true}  | ${17}
+    ${15}     | ${5}    | ${true}  | ${15}
+    ${19}     | ${6}    | ${true}  | ${3}
+  `(
+    '[useKeyboardMoveFocus] getNextFocusableOption should return next focusable option',
+    ({ currIndex, columns, infinite, result }) => {
+      const disabledDict = <Record<number, boolean>>{
+        0: true,
+        1: true,
+        2: true,
+        16: true,
+        19: true,
+      };
+
+      const { getNextFocusableOption } = _useKeyboardMoveFocus({
+        focusedIndexRef: renderHook(() => currIndex).result,
+        options,
+        columns,
+        infinite,
+        onFocusedIndexChange: (_) => {},
+        getOptionDisabled: (option) => !!disabledDict[option],
+      });
+      expect(getNextFocusableOption()).toEqual(result);
+    },
+  );
+});
+function renderHooks(
```

Comment:
remove

Reply:
ok! I did overlook.

Reply:
please review it whenever you get a chance. 
[582efed](https://github.com/ZouYouShun/Procyonidae-UI-Verify-Tool/pull/5/commits/582efed02e89cf17b76aad51847c3439e60634c2)